docs/ref/models/fields.txt

```diff
@@ -1215,6 +1215,47 @@ when :attr:`~django.forms.Field.localize` is ``False`` or
     information on the difference between the two, see Python's documentation
     for the :mod:`decimal` module.
 
+``GeneratedField``
+------------------
+
+.. versionadded:: 5.0
+
+.. class:: GeneratedField(expression, db_persist=None, output_field=None, **kwargs)
+
+A field which is computed based on other fields in the model. Similar to a
+database view, this field is managed and updated by the database itself. Uses
+the ``GENERATED`` SQL syntax.
+
+There are two kinds of generated columns: stored and virtual. A stored generated
+column is computed when it is written (inserted or updated) and occupies storage
+as if it were a normal column. A virtual generated column occupies no storage and
+is computed when it is read. Thus, a virtual generated column is similar to a view
+and a stored generated column is similar to a materialized view.
+
+.. attribute:: GeneratedField.expression
+
+    An :class:`Expression` used by the database to automatically set the field
+    value whenever the model is changed.
+
+    The expressions should be deterministic and only reference fields within
+    the model (in one database table). Generated fields cannot reference other
+    generated fields. Database backends can impose further restrictions.
+
+.. attribute:: GeneratedField.db_persist
```

Comment:
```suggestion
    generated fields. Database backends can impose further restrictions.

    .. admonition:: PostgreSQL
        The generation expression can only use immutable functions. This means that for example you will not be able to use the function of creating a JSON object because its implementation in PostgreSQL is not immutable

.. attribute:: GeneratedField.db_persist
```

Reply:
I tried to create a model with a generated field that contains a JSON built from the other fields of the model. I finally found out that the JSON build object function is not immutable in PostgreSQL. I guess it will be an idea that will come to many and so I have added this admonition.

Reply:
There are so many database-specific caveats/restrictions related to the generated fields that I don't want to open this can of worms and start documenting them, e.g. from [MySQL docs](https://dev.mysql.com/doc/refman/8.0/en/create-table-generated-columns.html), [SQLite docs](https://www.sqlite.org/gencol.html#limitations).

I'd add a general note:
```
.. admonition:: Database limitations

    There are many database-specific restrictions on generated fields that
    Django doesn't validate and the database may raise an error e.g. PostgreSQL
    requires functions and operators referenced in an index to be marked as
    ``IMMUTABLE`` .

    You should always check that  ``expression`` is supported on your database.
```

Reply:
Can we link to docs for the supported databases?

Reply:
Yes, added.